@here/harp-utils/lib/ContextLogger.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { IChannel } from "./Logger";
+
+export interface IContextLogger {
+    info(message?: any, ...optionalParams: any[]): void;
+    warn(message?: any, ...optionalParams: any[]): void;
+    error(message?: any, ...optionalParams: any[]): void;
+
+    pushAttr(name: string): void;
+    pushIndex(index: string | number): void;
+    pop(): void;
+}
+
+export class ContextLogger implements IContextLogger {
+    readonly context: string[] = [];
+    private m_headerLogged = false;
+    constructor(readonly m_logger: IChannel, readonly headerMessage: string) {}
+    pushAttr(name: string) {
+        this.context.push(`.${name}`);
+    }
+    pushIndex(index: string | number) {
+        this.context.push(`[${index}]`);
+    }
+    pop() {
+        this.context.pop();
+    }
+
+    // They, are public member functions it's just tslint who doesn't understand this.
+
+    // tslint:disable:member-ordering
+    warn = this.createLogMethod("warn");
```

Comment:
I can live with this JS style, but I would prefer TS, statically typed, tslintable code. Don't you?